File: Data Structures & Algorithms/eating-bananas/submission-2.java
class Solution {
    public int minEatingSpeed(int[] arr, int h) {
       int Left = 1;
       int Right = 0;
       for(int num : arr) {
        Right = Math.max(Right, num);
       }
      
       int Ans = 0;
       while(Left <= Right) {
        int Mid = Left + (Right - Left) / 2;
         int TotalHours = 0;
        for(int num : arr) {
            TotalHours +=(int) Math.ceil((double) num / Mid);
        }
        if(TotalHours <= h) {
            Ans = Mid;
           Right = Mid - 1;
        }
        else {
            Left = Mid + 1;
        }
       }
       return Ans;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    findMin(arr) {
        let Left = 0;
        let Right = arr.length - 1;
        while(Left < Right) {
            let Mid = Math.floor((Left + Right) / 2);
            if(arr[Mid] < arr[Right]) {
                Right = Mid;
            }
            else {
                Left = Mid + 1;
            }
        }
        return arr[Left];
    }
}
